### PR TITLE
Implement api,validation,state,device_mismatched: Part Ⅰ

### PR DESCRIPTION
**Pull request by haoxli**
Complete *.spec.ts in src/webgpu/api/validation and src/webgpu/api/validation/encoding

<hr>

**Author checklist for test code/plans:**

- [x] All outstanding work is tracked with "TODO" in a test/file description or `.unimplemented()` on a test.
- [x] New helpers, if any, are documented using `/** doc comments */` and can be found via `helper_index.txt`.
- [x] (Optional, sometimes not possible.) Tests pass (or partially pass without unexpected issues) in an implementation. (Add any extra details above.)

**[Reviewer sign-off](https://github.com/gpuweb/cts/blob/main/docs/reviews.md) for test code/plans:** (Note: feel free to pull in other reviewers at any time for any reason.)

- [x] The test path is reasonable, the [description](https://github.com/gpuweb/cts/blob/main/docs/intro/plans.md) "describes the test, succinctly, but in enough detail that a reader can read only the test plans in a file or directory and evaluate the completeness of the test coverage."
- [x] Tests appear to cover this area completely, except for outstanding TODOs. Validation tests use control cases.
    (This is critical for coverage. Assume anything without a TODO will be forgotten about forever.)
- [ ] Existing (or new) test helpers are used where they would reduce complexity.
- [ ] TypeScript code is readable and understandable (is unobtrusive, has reasonable type-safety/verbosity/dynamicity).

### CODE REVIEW

File: src/webgpu/api/validation/validation_test.ts [hunk 1]
```diff
@@ -292,6 +293,33 @@ export class ValidationTest extends GPUTest {
...
+  getDeviceMismatchedShaderModule(descriptor: GPUShaderModuleDescriptor): GPUShaderModule {
+    return this.mismatchedDevice.createShaderModule(descriptor);
+  }
+
+  /** Return a GPURenderBundleEncoder with descriptor from mismatched device. */
+  getDeviceMismatchedRenderBundle(
+    descriptor: GPURenderBundleEncoderDescriptor
+  ): GPURenderBundleEncoder {
+    return this.mismatchedDevice.createRenderBundleEncoder(descriptor);
+  }
```
Comment (kainino0x) [inline src/webgpu/api/validation/validation_test.ts:316]: see above, I don't think these 4 are needed.

File: src/webgpu/api/validation/validation_test.ts [hunk 2]
```diff
@@ -292,6 +293,33 @@ export class ValidationTest extends GPUTest {
...
+
+  /** Return a GPUQuerySet with descriptor from mismatched device. */
+  getDeviceMismatchedQuerySet(descriptor: GPUQuerySetDescriptor): GPUQuerySet {
+    return this.trackForCleanup(this.mismatchedDevice.createQuerySet(descriptor));
+  }
```
Comment (kainino0x) [inline src/webgpu/api/validation/validation_test.ts:321]: Probably not really necessary either; the trackForCleanup could just be skipped as the tests aren't creating large numbers of these query set objects.

**Review by kainino0x — APPROVED** [review]
LGTM, just ~one comment; please land after addressing comment

File: src/webgpu/api/validation/createComputePipeline.spec.ts [hunk 3]
```diff
@@ -180,11 +176,52 @@ g.test('pipeline_layout,device_mismatch')
...
+    const { isAsync, mismatched } = t.params;
+
+    if (mismatched) {
+      await t.selectMismatchedDeviceOrSkipTestCase(undefined);
+    }
+
+    const layoutDescriptor = { bindGroupLayouts: [] };
+    const layout = mismatched
+      ? t.getDeviceMismatchedPipelineLayout(layoutDescriptor)
+      : t.device.createPipelineLayout(layoutDescriptor);
```
[the remark this answers is not in the document]
Comment (haoxli) [reply src/webgpu/api/validation/createComputePipeline.spec.ts]: Agree with you! 

I think we should firstly support getting multiple devices from device pool, so we can avoid adding `selectMismatchedDeviceOrSkipTestCase ` and changing the name of `mismatchedDevice ` (maybe it should be named `secondaryDevice`) in the latter cases. I will do it in another patch.

**Pull request merged**